src/main/java/org/motechproject/newebodac/repository/custom/impl/FieldConfigRepositoryImpl.java

```diff
@@ -0,0 +1,41 @@
+package org.motechproject.newebodac.repository.custom.impl;
+
+import java.util.List;
+
+import javax.persistence.Entity;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.ParameterExpression;
+import javax.persistence.criteria.Root;
+
+import org.motechproject.newebodac.domain.FieldConfig;
+import org.motechproject.newebodac.domain.enums.EntityType;
+import org.motechproject.newebodac.dto.BaseDto;
+import org.motechproject.newebodac.exception.NewEbodacException;
+import org.motechproject.newebodac.repository.custom.FieldConfigRepositoryCustom;
+import org.motechproject.newebodac.repository.FieldConfigRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class FieldConfigRepositoryImpl extends BaseRepositoryImpl implements FieldConfigRepositoryCustom {
+  /**
+   * Finds entities matching all of the provided parameters. If there are no
+   * parameters, return null.
+   */
+  public BaseDto search(EntityType entity, BaseDto value) throws NewEbodacException {
+    CriteriaBuilder builder = entityManager.getCriteriaBuilder();
+
```

Comment:
I think first you need to fetch all the field configs for the entity and filer the list to find out which fields are unique.
You want to check the value only for unique fields, you will have the field name and type from the field config, this can be used to build the query.